### PR TITLE
Second part of Quarkus 3.13 upgrade: convert ITs to use @WithTestResource

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Completes Quarkus 3.9 -> 3.13 upgrade by converting use of `@QuarkusTestResource` (deprecated) into `@WithTestResource` (replacement).

Unfortunately turns out we have to use `restrictToAnnotatedClass = false` to specify old behavior, due to the way out ITs are structured with Nested classes (bad idea in hindsight).

**Which issue(s) this PR fixes**:
N/A

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by tatu-at-datastax** [comment]
Ok, unit tests now pass with added test resource inclusions.

Many ITs fail, but hopefully they are due to same root cause.

**Comment by tatu-at-datastax** [comment]
So about half of ITs fail:

```
Error:  Tests run: 159, Failures: 63, Errors: 25, Skipped: 3
```

and all seem to be due to namespace mismatch: either test namespace not created, or one created and one test uses does not match.
So solving that problem should possibly fix everything.

**Comment by jeffreyscarpenter** [comment]
@tatu-at-datastax how much work would be involved in refactoring the nested test classes?

**Comment by jeffreyscarpenter** [comment]
> isolation, while nice, would increase test runtime a lot, potentially.

Important point. Agree with your current approach.

File: src/test/java/io/stargate/sgv2/jsonapi/service/cqldriver/InvalidCredentialsTest.java [hunk 1]
```diff
@@ -30,6 +32,8 @@
 
 @QuarkusTest
 @TestProfile(InvalidCredentialsTest.TestProfile.class)
+// Since Quarkus 3.13, will not automatically get global test resources so need this:
+@WithTestResource(DseTestResource.class)
```
[the remark this answers is not in the document]
Comment (tatu-at-datastax) [reply src/test/java/io/stargate/sgv2/jsonapi/service/cqldriver/InvalidCredentialsTest.java:36]: Agreed.

**Pull request merged**